lib/spack/docs/spec_syntax.rst

```diff
@@ -0,0 +1,708 @@
+by a space. The command ``spack install libelf cppflags="-O3"%intel``
+will be interpreted as an attempt to set ``cppflags="-O3%intel"``.
+
+The six compiler flags are injected in the order of implicit make commands
+in GNU Autotools. If all flags are set, the order is
+``$cppflags $cflags|$cxxflags $ldflags <command> $ldlibs`` for C and C++, and
+``$fflags $cppflags $ldflags <command> $ldlibs`` for Fortran.
+
+-----------------------
+Architecture specifiers
+-----------------------
+
+Each node in the dependency graph of a spec has an architecture attribute.
+This attribute is a triplet of platform, operating system, and processor.
+You can specify the elements either separately by using
+the reserved keywords ``platform``, ``os``, and ``target``:
+
+.. code-block:: console
+
+   $ spack install libelf platform=linux
+   $ spack install libelf os=ubuntu18.04
+   $ spack install libelf target=broadwell
+
+Normally, users don't have to bother specifying the architecture if they
+are installing software for their current host, as in that case the
+values will be detected automatically. If you need fine-grained control
+over which packages use which targets (or over *all* packages' default
+target), see :ref:`package-preferences`.
+
+
+.. _support-for-microarchitectures:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Support for specific microarchitectures
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Spack knows how to detect and optimize for many specific microarchitectures and encodes this information in the ``target`` portion of the architecture specification.
+A complete list of the microarchitectures known to Spack can be obtained in the following way:
+
+.. command-output:: spack arch --known-targets
+
+When a spec is installed, Spack matches the compiler being used with the microarchitecture being targeted to inject appropriate optimization flags at compile time.
+Giving a command such as the following:
+
+.. code-block:: console
+
+   $ spack install zlib%gcc@14.2.0 target=icelake
+
+will produce compilation lines similar to:
+
+.. code-block:: console
+
+   $ /usr/bin/gcc-14 -march=icelake-client -mtune=icelake-client -c ztest10532.c
+   $ /usr/bin/gcc-14 -march=icelake-client -mtune=icelake-client -c -fPIC -O2 ztest10532.
+   ...
+
+where the flags ``-march=icelake-client -mtune=icelake-client`` are injected by Spack based on the requested target and compiler.
+
+If Spack knows that the requested compiler can't optimize for the current target
+or can't build binaries for that target at all, it will exit with a meaningful error message:
+
+.. code-block:: console
+
+   $ spack install zlib%gcc@5.5.0 target=icelake
+   ==> Error: cannot produce optimized binary for micro-architecture "icelake" with gcc@5.5.0 [supported compiler versions are 8:]
+
+When instead an old compiler is selected on a recent enough microarchitecture but there is
+no explicit ``target`` specification, Spack will optimize for the best match it can find instead
```

Comment:
```suggestion
Conversely, if an old compiler is selected for a newer microarchitecture, Spack will optimize for the best match it can find instead
```